dynamics.js

```diff
@@ -48,3 +48,201 @@ closerIcon.addEventListener('click', deactiveMenu);
+secondButtonAnchor.id = 'secondButtonAnchor';
+popupContainer1.id = 'popupContainer1'; // Background container
+popupContainer2.id = 'popupContainer2'; // Main container (with all child elements)
+cardButtonContainer.id = 'cardButtonContainer';
+popupCardButtonIcon1.id = 'popupCardButtonIcon1';
+popupCardButtonIcon2.id = 'popupCardButtonIcon2';
+
+//  Add classes and aditional characteristics
+popupParagraph.classList.add('text2');
+technologiesList.classList.add('ul-with');
+firstTechnology.classList.add('blue-li');
+secondTechnology.classList.add('blue-li');
+thirdTechnology.classList.add('blue-li');
+firstButton.classList.add('green-button');
+secondButton.classList.add('green-button');
+firstButton.innerText = 'Live version   ';
+secondButton.innerText = 'Source code   ';
+firstButtonAnchor.target = '_blank';
+secondButtonAnchor.target = '_blank';
+popupCloser.src = 'images/popupCloserIcon.svg';
+popupCardButtonIcon1.src = 'images/popupCardButtonIcon1.svg';
+popupCardButtonIcon2.src = 'images/popupCardButtonIcon2.svg';
+
+//  Assing information from objects and add them to the page
+function refreshPopup(currentObject) {
+  const [a, b, c] = currentObject.technologies;
+  projectImage.src = currentObject.image;
+  firstTechnology.innerText = a;
+  secondTechnology.innerText = b;
+  thirdTechnology.innerText = c;
+  popupTittle.innerText = currentObject.name;
+  popupParagraph.innerText = currentObject.description;
+  firstButtonAnchor.href = currentObject.live;
+  secondButtonAnchor.href = currentObject.source;
+  document.body.appendChild(popupContainer1);
+}
+
+//  Add elements to containers
+technologiesList.appendChild(firstTechnology);
+technologiesList.appendChild(secondTechnology);
+technologiesList.appendChild(thirdTechnology);
+firstButton.appendChild(popupCardButtonIcon1);
+secondButton.appendChild(popupCardButtonIcon2);
+firstButtonAnchor.appendChild(firstButton);
+secondButtonAnchor.appendChild(secondButton);
+cardButtonContainer.appendChild(firstButtonAnchor);
+cardButtonContainer.appendChild(secondButtonAnchor);
+popupContainer2.appendChild(projectImage);
+popupContainer2.appendChild(popupCloser);
+popupContainer2.appendChild(popupTittle);
+popupContainer2.appendChild(technologiesList);
+popupContainer2.appendChild(popupParagraph);
+popupContainer2.appendChild(cardButtonContainer);
+popupContainer1.appendChild(popupContainer2);
+
+//  Click actions to the work card buttons
+firstCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[0]);
+});
+
+secondCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[1]);
+});
+
+thirdCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[2]);
+});
+
+fourthCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[3]);
+});
+
+fifthCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[4]);
+});
+
+sixthCardButton.addEventListener('click', () => {
+  refreshPopup(masterArray[5]);
+});
```

Comment:
- [ ] This is a hard coded approach to the conditional rendering of the modal, a better approach would be to have the id of the modal passed in from the `index.htm` file, each button calling the same function buth with a different ID as argument.